### DISCUSSION TITLE
Add Data property to Avalonia.Media.Imaging.IImage

### DISCUSSION THREAD

**Discussion by PieroCastillo** [AvaloniaUI/Avalonia#5702] [Ideas]
Hi, I've a question, is there possible add a Data property(as Stream or byte[]) to IImage, now I'm making a BlurryImage control (temporal name), but I found a problem (for me), there are two ways to do it only:
1, Use WriteableBitmap: Maybe seems the best option, but has a problem, if I wanna use DrawingImage or similar(derivates of IImages) this option is imposible, also if that is posible to render it takes to long(update big writeablebitmaps takes too long) also is CPU dependent.
2. Use SkiaSharp: In my opinion is the best option, 'cause render using GPU, and it's more fast(render bigs images takes less than 1 ms), but this has a problem, needs or a Stream or a File, with IImage this is impossible, IImage has Size & Draw method only.

In a nutshell, is it possible implement a Data property as Stream/byte array in IImage? this could be an advantage for Avalonia, too.

**Comment by Gillibald** [comment]
This could be added to `IBitmap` https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Visuals/Media/Imaging/IBitmap.cs because IImage is just some general abstraction that is able to embed every kind of image like textures, drawings, etc.

**Reply by PieroCastillo** [reply]
> This could be added to `IBitmap` https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Visuals/Media/Imaging/IBitmap.cs because IImage is just some general abstraction that is able to embed every kind of image like textures, drawings, etc.

Is there a way to get the data of Bitmap or WriteableBitmap now?

**Reply by jp2masa** [reply]
I think you can do this:

```cs
using var fb = writeableBitmap.Lock();
// fb.Address is a pointer to the bitmap data
```

**Reply by PieroCastillo** [reply]
I found an easy way, just accept Bitmap and Save it to a Stream, next render it.
